### PR TITLE
Profile Model Created

### LINKED ISSUE #5: BE: Profile Model
Create Model for Profile

Profile (refer to mockup for required fields)
\- We don't need gender or birth date

### PR DESCRIPTION
**Pull request by abhkoirala**
### What this PR does (required):
- This PR contains the model for Profile

### Screenshots / Videos (required):
- 
image

### Any information needed to test this feature (required):
- NA

### Any issues with the current functionality (optional):
- NA

### CODE REVIEW

File: server/models/Profile.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    first_name: {
+        type: String,
+        required: true
+      },
+      last_name: {
+        type: String,
+        required: true
+      }
+    }
+],
```
Comment (g4rry420) [inline server/models/Profile.js]: I am seeing a couple indentation a few lines.

File: server/models/Profile.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
+
+const userSchema = new mongoose.Schema({
+    user:{
+        type: mongoose.Schema.Types.ObjectId,
+        ref:'user'
+    },
+    name: [
```
Comment (g4rry420) [inline server/models/Profile.js]: I think we don't need a separate `name` array field for this. But, Let me know what other team leads think. :)

File: server/models/Profile.js [hunk 3]
```diff
@@ -0,0 +1,41 @@
+const mongoose = require("mongoose");
+const bcrypt = require("bcryptjs");
```
Comment (g4rry420) [inline server/models/Profile.js]:
```suggestion
```
I believe we don't need this line.

File: server/models/Profile.js [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    type: String,
+    required: true
+  },
+  register_date: {
+    type: Date,
+    default: Date.now
+  }
+
+    
+});
```
Comment (g4rry420) [inline server/models/Profile.js:24]:
```suggestion
}, { timestamps: true });
```
You can pass this field as a second object to Schema. This creates two fields `createdAt` and `updatedAt`. So, we don't need `register_date` field.

**Review by g4rry420 — COMMENTED** [review]
All looks goods. Just a few suggestions. Don't approve them yet. Let the team leads review them first.

**Review by bonnieli — COMMENTED** [review]
@g4rry420's code review is great - looks like just some linting issues you'd need to make sure to fix

**Comment by yliu298** [comment]
hi @abhkoirala , I am working on profile controller and review controller, they both need to use profile model, and I believe the profile model will be used at user and dog sitter, so I have some suggestions on this model, please kindly check the code below: 
const mongoose = require("mongoose");

const proflieSchema = new mongoose.Schema({
    firstName: {
      type: String,
      required: true,
    },
    lastName: {
      type: String,
      required: true,
    },
    phoneNumber: {
      type: String,
      required: true,
    },
    address: {
      type: String,
      required: true,
    },
    img: {
      data: Buffer,
      contentType: String
    },
    description: String,
    isDogSitter: {
      type: Boolean,
      required: true,
      default: false,
    },
    isAvailable: {
      type: Boolean,
      required: true,
      default: true,
    }
})

module.exports = Profile = mongoose.model("profile", proflieSchema);

**Comment by abhkoirala** [comment]
@yliu298 @g4rry420 @bonnieli I made the suggested changes. Do I need to submit a new PR or its fine? Do suggest if any more changes are needed. Your suggestions are valuable

**Comment by bonnieli** [comment]
Hi @abhkoirala a couple of suggestions and comments

1. Looks like there's still one comment unresolved made by @g4rry420 you'd like to address
2. Looks like still some linting (spacing) issues
3. Just an advice going forward, we aren't able to view the code properly so if you'd like to share a code block, you can use markdown syntax
4. No need to make another PR, as you may notice in this PR, your changes have been reflected

**Comment by abhkoirala** [comment]
@bonnieli I made the timestamp change that was needed.

**Review by bonnieli — APPROVED** [review]
still some spacing issues left

Recommended to just download the proper eslint and prettier extensions, the config is included in the repo!

**Pull request merged**